src/variables/components/TypeAheadVariableDropdown.tsx

```diff
@@ -51,107 +56,184 @@ class TypeAheadVariableDropdown extends PureComponent<Props, MyState> {
       menuOpen: null,
       loaded: false,
     }
+
+    // if it's a csv var, loading is instantaneous, so if it is done just set it:
+    if (props.status === RemoteDataState.Done) {
+      defaultState.typedValue = props.selectedValue
+      defaultState.actualVal = props.selectedValue
+      defaultState.loaded = true
+    }
+
+    this.state = defaultState
   }
 
   // set the 'shownValues' after loading, and
   // resets the menuOpen variable
   componentDidUpdate(prevProps, prevState) {
     const prevVals = prevProps.values
-    const {values, selectedValue} = this.props
+    const {values, selectedValue, status} = this.props
+    const {status: prevStatus} = prevProps
     const {actualVal, loaded, selectHappened, menuOpen} = this.state
     const {
       actualVal: prevActualVal,
       selectHappened: prevSelectHappened,
       menuOpen: prevMenuOpen,
     } = prevState
 
+    let newState = {}
+
+    const justLoaded = () =>
+      status === RemoteDataState.Done && prevStatus !== RemoteDataState.Done
+
+    const justSelected = () =>
+      selectHappened &&
+      !prevSelectHappened &&
+      actualVal &&
+      actualVal !== prevActualVal
+
     // this is for updating the values:
     // (only want this to run *once* when the values get loaded)
-    if (!loaded && prevVals.length !== values.length) {
-      this.setState({
+    if (justLoaded()) {
+      newState = this.getInitialValuesAfterLoading()
+    } else if (
+      !loaded &&
+      status === RemoteDataState.Done &&
+      prevVals.length !== values.length
+    ) {
+      newState = {
         shownValues: values,
         typedValue: selectedValue,
         loaded: true,
-      })
+      }
     }
 
-    // unset the menuOpen; it should be set to closed (or open) only once; then undone
+    /**
+     * unset the menuOpen; it should be set to closed (or open) only once; then undone
+     * this is needed because: (from clockface Dropdown.tsx documentation):
+     * if the string is set to 'open', and then the user closes it, and the code sets it to open again,
+     * unless the code sets it to something else in between (like null or 'close'),
+     * then nothing will happen- the menu will not open)
+     */
+
     if (menuOpen !== prevMenuOpen && menuOpen !== null) {
-      this.setState({menuOpen: null})
+      newState['menuOpen'] = null
     }
 
     // need to have this, as the 'onClickAwayHere' gets triggered *before*
     // the selected value is set to the actualValue (it keeps re-using the original
     // property)
```

Comment:
is this comment referring to anything that you can see? i don't know what it's referring to. let's kill it, or move it to where it's relevant